synapse/lib/stormtypes.py

```diff
@@ -7814,24 +7819,20 @@ async def getStorNodesByForm(self, form):
     @stormfunc(readonly=True)
     async def getStorNodesByProp(self, propname, propvalu=None, propcmpr='='):
         propname = await tostr(propname)
-        propvalu = await tostor(propvalu)
-        propcmpr = await tostr(propcmpr)
-
-        layriden = self.valu.get('iden')
-        await self.runt.reqUserCanReadLayer(layriden)
-        layr = self.runt.snap.core.getLayer(layriden)
+        propvalu = await toprim(propvalu)
 
-        prop = self.runt.snap.core.model.reqProp(propname)
+        prop = self.runt.snap.core.model.prop(propname)
+        if prop is None:
+            mesg = f'The property {propname} does not exist.'
+            raise s_exc.NoSuchProp(mesg=mesg)
 
-        if propvalu is not None:
-            norm, info = prop.type.norm(propvalu)
-            cmprvals = prop.type.getStorCmprs(propcmpr, norm)
-            async for _, buid, sode in layr.liftByPropValu(prop.form.name, prop.name, cmprvals):
-                yield (s_common.ehex(buid), sode)
+        if prop.isform and propvalu is None:
+            async for buid, sode in self.getStorNodesByForm(propname):
```

Comment:
I think this was supposed to lift nodes where the ndef was in the layer, not the `getStorNodesByForm` behavior?

Reply:
I thought getStorNodesByForm() did that? What's the difference?